x-pack/plugin/ql/src/test/java/org/elasticsearch/xpack/ql/querydsl/query/BoolQueryTests.java

```diff
@@ -35,15 +37,15 @@ public void testEqualsAndHashCode() {
     }
 
     private static BoolQuery copy(BoolQuery query) {
-        return new BoolQuery(query.source(), query.isAnd(), query.left(), query.right());
+        return new BoolQuery(query.source(), query.isAnd(), left(query), right(query));
     }
 
     private static BoolQuery mutate(BoolQuery query) {
         List<Function<BoolQuery, BoolQuery>> options = Arrays.asList(
```

Comment:
This test doesn't fully reflect the changes in this PR, but I'm ok with it.

Reply:
Fixed it nevertheless.